string.h

```diff
@@ -1,49 +1,73 @@
-//lang::CwC
+// lang::CwC
 #pragma once
 
 #include "object.h"
+#include <cstdio>
 #include <cstdlib>
 #include <cstring>
-#include <cstdio> 
 
 /**
- * An immutable String class representing a char*
- * author: chasebish */
+ * A String class representing a char[]
+ * author: chasebish
+ */
 class String : public Object {
 public:
-  /** CONSTRUCTORS & DESTRUCTORS **/
-
-  /* Creates a String copying s */
-  String(const char* s);
+  /**
+   * Creates a String by copying a string literal.
+   *
+   * @arg s string literal to be used as a source for this String object
+   */
+  String(const char *s);
 
-  /* Copies a String copying the value from s */
-  String(String* const s);
+  /**
+   * Copy constructor.
+   *
+   * @arg s source String object to be copied from
+   */
+  String(const String *s);
 
-  /* Clears String from memory */
   ~String();
 
+  /**
+   * Inherits the following from parent Object:
+   *
+   * bool equals(const Object *o) const;
+   * size_t hash() const;
+   */
 
-  /** INHERITED METHODS **/
-
-  /* Inherited from Object, generates a hash for a String */
-  size_t hash();
-
-  /* Inherited from Object, checks equality between an String and an Object */
-  bool equals(Object* const obj);
-
+  /**
+   * Compares strings based on alphabetical order according to the following
+   * ruleset:
+   *   < 0 -> this String is less than String s
+   *   = 0 -> this String is equal to String s
+   *   > 0 -> this String is greater than String s
+   *
+   * @arg s  the String to compare this against
+   * @return an int resulting from applying the above rules
+   */
+  int cmp(const String *s) const;
```

Comment:
I would love to rename this, `compare` seems short enough. Was actually going to create a follow up PR to move compare to `Object` though because pushing `sort` behavior into the base `Object` implementation of List/Array/Vector allows for sorting lists of any type. e.g. `Number`, `StringWCount`(WU3)...
Missing a Comparable interface but don't really want to implement a ComparableList on top of ObjectList...

Reply:
Renamed